vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java

```diff
@@ -122,7 +126,57 @@ public GlobalCommandTarget getTarget() {
   private boolean passesPropertyFastMatch(GamePiece gamePiece) {
     if (!target.fastMatchProperty || fastProperty.isEmpty()) return true;
     final String value = (String) gamePiece.getProperty(fastProperty);
-    return fastValue.equals(value);
+
+    // Intentionally favors the default "Equals" as first to process
+    switch (target.targetCompare) {
+    case EQUALS:
+      return fastValue.equals(value);
+    case NOT_EQUALS:
+      return !fastValue.equals(value);
+    case MATCH:
+      return fastPattern.matcher(value).matches();
+    case NOT_MATCH:
+      return !fastPattern.matcher(value).matches();
+    }
+
+    // Lexical comparisons for strings
+    if (!fastIsNumber || !isNumeric(value)) {
+      switch (target.targetCompare) {
+      case GREATER_EQUALS:
+        return value.compareTo(fastValue) >= 0;
+      case GREATER:
+        return value.compareTo(fastValue) > 0;
+      case LESS_EQUALS:
+        return value.compareTo(fastValue) <= 0;
+      case LESS:
+        return value.compareTo(fastValue) < 0;
+      }
+    }
+
+    // Numerical comparisons for numbers
+    final double num = Double.parseDouble(value);
+
+    switch (target.targetCompare) {
+    case GREATER_EQUALS:
+      return num >= fastNumber;
+    case GREATER:
+      return num > fastNumber;
+    case LESS_EQUALS:
+      return num <= fastNumber;
+    case LESS:
+      return num < fastNumber;
+    }
+
+    return false; // Never gets here, but checkStyle doesn't understand that.
+  }
+
+  /**
+   * Need a super-fast (i.e. not dependent on exception-throwing) plan for detecting valid numbers
+   * @param s string to check
+   * @return true if a value number
+   */
+  private boolean isNumeric(String s) {
+    return s.matches("(\\+-)?\\d+(\\.\\d+)?");  //match a number with optional +/- and decimal.
```

Comment:
It's always faster to use a precompiled pattern.

Reply:
Good point - AND HERE I JUST LEARNED HOW TO DO THAT!